pkg/cmd/p2p/promote/p2p_promote.go

```diff
@@ -0,0 +1,183 @@
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.ImageWithTag = args[0]
+
+			opts.Streams = userio.NewIOStreams(
+				cmd.InOrStdin(),
+				cmd.OutOrStdout(),
+			)
+
+			return run(&opts)
+		},
+	}
+
+	requiredFlagMap := map[string]*string{
+		"source-registry": &opts.SourceRegistry,
+		"source-stage":    &opts.SourceStage,
+		"dest-registry":   &opts.DestRegistry,
+		"dest-stage":      &opts.DestStage,
+	}
+
+	for name, field := range requiredFlagMap {
+		err := addFlag(promoteCommand, field, name, true)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	optionalFlagMap := map[string]*string{
+		"source-auth-override": &opts.SourceAuthOverride,
+		"dest-auth-override":   &opts.DestAuthOverride,
+	}
+	for name, field := range optionalFlagMap {
+		err := addFlag(promoteCommand, field, name, false)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return promoteCommand, nil
+}
+
+func addFlag(promoteCommand *cobra.Command, field *string, name string, required bool) error {
+	envVariableName := strings.ToUpper(strings.ReplaceAll(name, "-", "_"))
+
+	description := fmt.Sprintf("optional, defaults to environment variable: %s", envVariableName)
+	if required {
+		description = fmt.Sprintf("required, defaults to environment variable: %s", envVariableName)
+	}
+
+	promoteCommand.Flags().StringVar(
+		field,
+		name,
+		"",
+		description,
+	)
+
+	envVariableValue := os.Getenv(envVariableName)
+
+	if envVariableValue != "" {
+		*field = envVariableValue
+	}
+
+	if envVariableValue == "" && required {
+		if err := promoteCommand.MarkFlagRequired(name); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func run(opts *promoteOpts) error {
+
+	sourceImage := &imageOpts{
+		ImageNameWithTag: opts.ImageWithTag,
+		Registry:         opts.SourceRegistry,
+		RepoPath:         opts.SourceStage,
+		AuthOverride:     opts.SourceAuthOverride,
+	}
+
+	destinationImage := &imageOpts{
+		ImageNameWithTag: opts.ImageWithTag,
+		Registry:         opts.DestRegistry,
+		RepoPath:         opts.DestStage,
+		AuthOverride:     opts.DestAuthOverride,
+	}
+
+	logInfo := opts.Streams.Info
+	logInfo("Configuring docker with gcloud")
+	output, err := configureDockerWithGcloud()
+	if err != nil {
+		return err
+	}
+	logInfo(string(output))
+
+	logInfo("Pulling image", imageUri(sourceImage))
+	output, err = pullDockerImage(sourceImage)
+	if err != nil {
+		return err
+	}
+	logInfo(string(output))
+
+	logInfo("Tagging image", imageUri(sourceImage), "with", imageUri(destinationImage))
+	output, err = tagDockerImage(sourceImage, destinationImage)
+	if err != nil {
+		return err
+	}
+	logInfo(string(output))
+
+	logInfo("Pushing image", imageUri(destinationImage))
+	output, err = pushDockerImage(destinationImage)
+	if err != nil {
+		return err
+	}
+	logInfo(string(output))
+
+	return nil
+}
+
+func configureDockerWithGcloud() ([]byte, error) {
+	return exec.Command("gcloud", "auth", "configure-docker", "--quiet", "europe-west2-docker.pkg.dev").Output()
```

Comment:
In `env connect` command we are checking for required dependencies to be installed:
https://github.com/coreeng/corectl/blob/e5f2ac8ef2f547f6d1a03a0c38d9f5384c280a7b/pkg/env/validate.go#L39
I think it would be nice to do similar thing here. + maybe unify the interface for this

Reply:
make sense, I've reused this funciton